internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -149,55 +138,3 @@ func (h *serverSidePatchHelper) Patch(ctx context.Context) error {
 	}
 	return h.client.Patch(ctx, h.modified, client.Apply, options...)
 }
-
-// cleanupLegacyManagedFields cleanups managed field management in place before introducing SSA.
-// NOTE: this operation can trigger a machine rollout, but this is considered acceptable given that ClusterClass is still alpha
-// and SSA adoption align the topology controller with K8s recommended solution for many controllers authoring the same object.
-func cleanupLegacyManagedFields(ctx context.Context, obj *unstructured.Unstructured, c client.Client) error {
```

Comment:
@fabriziopandini the test `Server side apply cleanups legacy managed fields` expects this to happen, should it be kept without the annotation check? or this cleanup is not needed anymore?

Reply:
I think we can drop the entire test it seems to exlusively validate that the legacy annotation is gone + managed fields have been adjusted correctly